File: 666-2/D.cpp
#include <cstdio>
#include <vector>
using namespace std;

int t, a, last, m, n, p;
vector<int> as;

bool play() {
    bool turn = false;
    last = -1;
    while (true) {
        p = 0;
        m = 0;
        for (int i = 0; i < as.size(); ++i) {
            if (i != last) {
                if (as[i] > m) {
                    m = as[i];
                    p = i;
                }
            }
        }
        if (m == 0) {
            return turn;
        }
        last = p;
        as[last] -= 1;
        turn = !turn;
    }
    return turn;
}

int main() {
    scanf("%d", &t);
    for (int i = 0; i < t; ++i) {
        scanf("%d", &n);
        as.clear();
        for (int j = 0; j < n; ++j) {
            scanf("%d", &a);
            as.emplace_back(a);
        }
        printf(play() ? "T\n" : "HL\n");
    }
}

File: 666-2/D.py
t = int(input())

def play(a):
    turn = True
    while True:
        last = -1
        m = 0
        for i in range(len(a)):
            if i != m:
                if a[i] > m:
                    m = a[i]
                    last = i
        if last == -1:
            return turn
        a[last] -= 1            
        turn = not turn
    return turn

for i in range(t):
    n = int(input())
    a = list(map(int, input().split()))
    print("T" if play(a) else "HL")